crates/polars-arrow/src/legacy/kernels/take_agg/mod.rs

```diff
@@ -23,17 +23,15 @@ pub unsafe fn take_agg_no_null_primitive_iter_unchecked<
     arr: &PrimitiveArray<T>,
     indices: I,
     f: F,
-    init: TOut,
 ) -> TOut {
```

Comment:
Can you make this return an `Option<TOut>` instead of `.expect` for empty input?

Reply:
From what I can tell, `TOut` is only different from `T` when calculating the group mean for non-float types, where the values are converted to `f64`s before being summed up. However, float types are converted to `f64` after summing the values but before dividing by the count. While the order doesn't matter for `f64`s, it risks an overflow for `f32`s. Is that the intended behavior, or should I change `TOut` to `f64` in those cases?

Reply:
Here's an example where the `mean` behavior seems weird. Specifically, `polars` calculates the mean of the same data differently depending on whether it was part of a group by or not.
```python
import polars as pl

big_float = 2.0e38

df = pl.DataFrame(
    [
        pl.Series("groups", ["a", "a"]),
        pl.Series("values", [big_float, big_float], dtype=pl.Float32),
    ]
)

df.mean()["values"] # correctly gives 2.0e38

df.group_by("groups").mean()["values"] # inf
```

It's a small fix here, or I can cut another ticket and leave it for a later PR if you'd prefer.